docs/docs/contributions/development/maintenance-tasks-and-scripts.mdx

```diff
@@ -0,0 +1,86 @@
+---
+    title: Maintenance tasks and scripts
+    sidebar_position: 6
+---
+
+There's a variety of maintenance tasks that happen on different frequencies.
+
+## Update the default/known versions of a built-in tool/subsystem
+
+### External tools (downloaded executables)
+
+Some tools use [the `ExternalTool` class](../../writing-plugins/the-rules-api/installing-tools#externaltool-install-pre-compiled-binaries) to download a binary from the internet, verify its size and hash, and then execute it.
+
+To update these:
+
+1. For each platform and version to add:
+   1. Download the archive/binary.
+   2. Verify it: check signatures and/or hashes if available.
+   3. Compute the sha256 hash and byte length. For example, if it's called `archive.zip`: `tee >(shasum -a 256) >(wc -c) > /dev/null < archive.zip`.
```

Comment:
I really like the one line approach with `curl`, such as:
https://github.com/pantsbuild/pants/blob/e44697e5dc57e80c91ba7396f46fe044ea40ba40/src/python/pants/core/target_types.py#L225-L228

Reply:
I considered that style, but I thought step 2 was fairly important: if Pants distributed an obviously-incorrect artifact by default (e.g. someone has uploaded a different binary that doesn't match signature), that would be very bad behaviour on our part! 

It seems more acceptable/less egregious if a user does it to themselves, in their own local config.